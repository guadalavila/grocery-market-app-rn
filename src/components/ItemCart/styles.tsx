import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.white,
        borderRadius: Sizes.S,
        paddingHorizontal: Sizes.L,
        padding: Sizes.M,
        marginVertical: Sizes.S,
        marginHorizontal: Sizes.L,
        flexDirection: 'row',
    },
    image: {
        width: 100,
        height: 70,
    },
    containerDescription: {
        paddingLeft: Sizes.M,
        flex: 1,
    },
    title: {
        fontWeight: '900',
        fontSize: 16,
        color: Colors.black,
    },
    price: {
        paddingTop: Sizes.M,
        textAlign: 'right',
        fontSize: 18,
        color: Colors.black,
        fontWeight: '900',
    },
});
