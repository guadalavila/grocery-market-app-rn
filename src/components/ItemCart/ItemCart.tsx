import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Product } from '../../models/Product';
import { GlobalStyles } from '../../utils/GlobalStyles';
import { styles } from './styles';

interface IItemCartProps {
    product: Product;
}

const ItemCart = ({ product: { image, name, price } }: IItemCartProps) => {
    return (
        <View style={styles.container}>
            <Image resizeMode='center' style={styles.image} source={{ uri: image }} />
            <View style={styles.containerDescription}>
                <View style={[GlobalStyles.row, GlobalStyles.justBtw]}>
                    <Text style={styles.title}>{name}</Text>
                    <TouchableOpacity activeOpacity={0.7} onPress={() => {}}>
                        <Icon name='close' size={24} />
                    </TouchableOpacity>
                </View>
                <Text style={styles.price}>${price}.00</Text>
            </View>
        </View>
    );
};

export default ItemCart;
