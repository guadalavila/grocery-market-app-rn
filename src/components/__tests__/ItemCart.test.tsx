import { render, screen } from '@testing-library/react-native';
import React from 'react';
import ItemCart from '../ItemCart/ItemCart';

describe('test ItemCart component', () => {
    const itemMock = {
        product: {
            id: '6b17d300-943a-11ed-a1eb-0242ac120002',
            image: 'https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp',
            name: 'Banana',
            description: 'Organicas',
            price: 20.0,
        },
        quantity: 2,
    };
    const onPressMock = jest.fn();

    test('should match snapshot', () => {
        const component = render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(component).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(screen).toBeTruthy();
    });

    test('should render image', () => {
        render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(screen.getByTestId('image')).toBeTruthy();
        expect(screen.getByTestId('image').props.resizeMode).toBe('center');
    });

    test('should render close icon', () => {
        render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(screen.getByTestId('icon-close')).toBeTruthy();
        expect(screen.getByTestId('icon-close').props.name).toBe('close');
        expect(screen.getByTestId('icon-close').props.size).toBe(24);
    });

    test('should render texts', () => {
        render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(screen.getByText(itemMock.product.name)).toBeTruthy();
        expect(screen.getByText(`$${itemMock.product.price}.00`)).toBeTruthy();
        expect(screen.getByText(`${itemMock.quantity.toString()} unidades`)).toBeTruthy();
    });

    test('should render button', () => {
        render(<ItemCart item={itemMock} onPressClose={onPressMock} />);
        expect(screen.getByTestId('close-button')).toBeTruthy();
    });
});
