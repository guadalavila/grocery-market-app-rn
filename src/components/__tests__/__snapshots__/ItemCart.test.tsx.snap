// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemCart component should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "flexDirection": "row",
      "marginHorizontal": 24,
      "marginVertical": 4,
      "padding": 8,
      "paddingHorizontal": 24,
    }
  }
>
  <Image
    resizeMode="center"
    source={
      Object {
        "uri": "https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 50,
        "width": 90,
      }
    }
    testID="image"
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingLeft": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "900",
          }
        }
      >
        Banana
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="close-button"
      >
        <MockedIoniconsIcons
          name="close"
          size={24}
          testID="icon-close"
        />
      </View>
    </View>
    <Text>
      2
       
      unidades
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "900",
          "textAlign": "right",
        }
      }
    >
      $
      20
      .00
    </Text>
  </View>
</View>
`;
